etl/dw/f_historico_geracao_distribuida.py

```
@@ -0,0 +1,203 @@
+
+    def delete(self):
+        self._conn_output.execute(f'''--sql
+            DELETE FROM {self.schema}.{self.table_name} f
+            WHERE f.dt_copia = TO_TIMESTAMP({self.dt_update.strftime('%d/%m/%Y')}, 'dd/mm/yyyy')
+        ;'''[5:-1])
+
+    def run(self):
+        self.before_run()
+        self.extract()
+        self.delete()
+        for self.df_extract in self.itercursor():
```

Ruff (0.12.2)

Loop control variable `self` overrides iterable it iterates

(B020)